deepchem/feat/dft_data.py

```diff
@@ -0,0 +1,338 @@
+"""
+Density Functional Theory Data
+Derived from: https://github.com/mfkasim1/xcnn/blob/f2cb9777da2961ac553f256ecdcca3e314a538ca/xcdnn2/entry.py
+"""
+from __future__ import annotations
+from abc import abstractmethod, abstractproperty
+from typing import List, Dict, Optional
+import numpy as np
+
+# dqc depend
+try:
+    import dqc
+    from dqc.system.mol import Mol
+    from dqc.system.base_system import BaseSystem
+    from dqc.grid.base_grid import BaseGrid
+    from deepchem.utils.dftutils import KSCalc
+except:
+    pass
+
+
+class DFTSystem():
+    """
+    The DFTSystem class creates and returns the various systems in an entry object as dictionaries.
+
+    Examples
+    --------
+    >>> from deepchem.feat.dft_data import DFTSystem
+    >>> systems = {'moldesc': 'Li 1.5070 0 0; H -1.5070 0 0',
+                'basis': '6-311++G(3df,3pd)'}
+    >>> output = DFTSystem(systems)
+
+    Returns
+    -------
+    DFTSystem object for all the individual atoms/ions/molecules in an entry object.
+
+    References
+    ----------
+    Kasim, Muhammad F., and Sam M. Vinko. "Learning the exchange-correlation functional from nature with fully differentiable density functional theory." Physical Review Letters 127.12 (2021): 126403.
+
+    https://github.com/diffqc/dqc/blob/0fe821fc92cb3457fb14f6dff0c223641c514ddb/dqc/system/base_system.py
+    """
+
+    def __init__(self, system: Dict):
+        self.system = system
+        self.moldesc = system["moldesc"]
+        self.basis = system["basis"]
+        """
+        Parameters
+        ----------
+        system: Dict
+            system is a dictionary containing information on the atomic positions,
+            atomic numbers, and basis set used for the DFT calculation.
+        """
+
+    def get_dqc_mol(self, pos_reqgrad: bool = False) -> BaseSystem:
+        """
+        This method converts the system dictionary to a DQC system and returns it.
+        Parameters
+        ----------
+        pos_reqgrad: bool
+            decides if the atomic position require gradient calculation.
+        Returns
+        -------
+        mol
+            DQC mol object
+        """
+        atomzs, atomposs = dqc.parse_moldesc(self.moldesc)
+        if pos_reqgrad:
+            atomposs.requires_grad_()
+        mol = Mol(self.moldesc, self.basis)
+        return mol
+
+
+class DFTEntry():
+    """
+    Handles creating and initialising DFTEntry objects from the dataset. This object contains information    about the various systems in the datapoint (atoms, molecules and ions) along with the ground truth
+    values.
+    Notes
+    -----
+    Entry class should not be initialized directly, but created through
+    ``Entry.create``
+
+    Example
+    -------
+    >>> from deepchem.feat.dft_data import DFTEntry
+    >>> e_type= 'dm'
+    >>> true_val= 'deepchem/data/tests/dftHF_output.npy'
+    >>> systems = [{
```

Comment:
Doctest is complaining here. I think you want to move these all onto one line. Try running doctest locally to check correctness